### PR TITLE
[Redux] Add plumbing for redux and sagas

### LINKED ISSUE #123: Setup Redux

### PR DESCRIPTION
**Pull request by LeonSo7**
Add plumbing for redux and sagas

Closes #123

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,9851 +1,8 @@
 {
   "name": "project-sayyara",
   "version": "0.1.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
Comment (arkinmodi) [inline package-lock.json:4]: Are you using node 16? Why are you downgrading the `package-lock.json`?

File: src/redux/store.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+export interface State {}
+
+export interface SagaStore extends Store {
+  sagaTask?: Task;
+}
+
+export const makeStore = (context: Context) => {
+  const sagaMiddleware = createSagaMiddleware();
+
+  const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));
```
Comment (arkinmodi) [inline src/redux/store.ts:16]: `createStore` is deprecated. Can we switch to something else?

File: src/redux/store.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { createStore, Store, combineReducers, applyMiddleware } from "redux";
```
Comment (arkinmodi) [inline src/redux/store.ts:1]: `combineReducers` is an unused import.

File: src/redux/store.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import { rootSaga } from "./sagas";
+import { rootReducer } from "./reducers";
+
+export interface State {}
+
+export interface SagaStore extends Store {
+  sagaTask?: Task;
+}
+
+export const makeStore = (context: Context) => {
```
Comment (arkinmodi) [inline src/redux/store.ts:13]:
```suggestion
export const makeStore = (_context: Context) => {
```

**Review by arkinmodi — COMMENTED** [review]
There is a console error on both client and server. Can you fix these or explain why there are there?
Client:
```
Store does not have a valid reducer. Make sure the argument passed to combineReducers is an object whose values are reducers. 
    at MyApp (webpack-internal:///./src/pages/_app.tsx:22:28)
    at ErrorBoundary (webpack-internal:///./node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:8:20742)
    at ReactDevOverlay (webpack-internal:///./node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:8:23635)
    at Container (webpack-internal:///./node_modules/next/dist/client/index.js:70:9)
    at AppContainer (webpack-internal:///./node_modules/next/dist/client/index.js:216:26)
    at Root (webpack-internal:///./node_modules/next/dist/client/index.js:407:27)
```

Server:
```
Store does not have a valid reducer. Make sure the argument passed to combineReducers is an object whose values are reducers.
```

**Comment by LeonSo7** [comment]
> There is a console error on both client and server. Can you fix these or explain why there are there? Client:
> 
> ```
> Store does not have a valid reducer. Make sure the argument passed to combineReducers is an object whose values are reducers. 
>     at MyApp (webpack-internal:///./src/pages/_app.tsx:22:28)
>     at ErrorBoundary (webpack-internal:///./node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:8:20742)
>     at ReactDevOverlay (webpack-internal:///./node_modules/next/dist/compiled/@next/react-dev-overlay/dist/client.js:8:23635)
>     at Container (webpack-internal:///./node_modules/next/dist/client/index.js:70:9)
>     at AppContainer (webpack-internal:///./node_modules/next/dist/client/index.js:216:26)
>     at Root (webpack-internal:///./node_modules/next/dist/client/index.js:407:27)
> ```
> 
> Server:
> 
> ```
> Store does not have a valid reducer. Make sure the argument passed to combineReducers is an object whose values are reducers.
> ```

This should just be a warning. Since we don't have any reducers yet, we will get this warning.

**Comment by arkinmodi** [comment]
Can you link the issue?

File: package-lock.json (hunk 1 above)
Comment (LeonSo7) [reply package-lock.json]: Fixed

File: src/redux/store.ts (hunk 2 above)
Comment (LeonSo7) [reply src/redux/store.ts]: Updated to use redux toolkit. Replaced with `configureStore`.

File: src/redux/store.ts (hunk 3 above)
Comment (LeonSo7) [reply src/redux/store.ts]: Updated

**Review by arkinmodi — APPROVED** [review]

**Pull request merged**